enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+  that I can avoid Amazon's new charge for public IP addresses.
+
+### Goals
+- Users are able to use EIP for a NLB Ingress Controller.
+- Users are able to create a NLB default Ingress Controller during install time with the EIPs specified during install.
+- The ELB of an existing IngressController should be able to be reconfigured to use specific EIPs by updating the IngressController CR and recreating the ELB with Load Balancer Type as `NLB`.
+
+### Non-Goals
+
+- Creation of EIPs in AWS.
+- Keep a track of changes in the EIP in the AWS environment.
+- Static IP usage with NLBs for API server.
+- Check the number of public subnets available and then compare number of subnets
+  with the number of EIPs the user provides.
+
+## Proposal
+
+EIP = AWS Elastic IP, NLB = AWS Network LoadBalancer 
+
+In this enhancement we would be adding API fields in the installer and the Ingress Controller CR
+to set the AWS EIP for the NLB Ingress Controller. The cluster ingress operator will then create 
+the ingress controller and get the EIP assigned to the service LoadBalancer with the annotation 
+`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+Note: `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` is a [kubernetes service annotation](https://kubernetes.io/docs/reference/labels-annotations-taints/#service-beta-kubernetes-io-aws-load-balancer-eip-allocations).
+
+
+### Workflow Description
+
+Pre-requisites: One must [allocate Elastic IP address](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating)
+in their AWS environment which could be then used in the `eipAllocations` field of the IngressController CR.
+Note: The AWS account has some cap on the number of EIPs per region. By default, all AWS accounts have a quota of five (5) Elastic IP addresses per Region, because public (IPv4) 
+internet addresses are a scarce public resource. For more details please check [Elastic IP address quota](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-limit)
+When this cap is hit you will get the following error: `Elastic IP address could not be allocated.
+The maximum number of addresses has been reached.`
+
+**cluster administrator** is a human user responsible for deploying a
+cluster. A cluster administrator also has the rights to modify the cluster level components.
+
+#### Set EIP using an installer:
+1. Cluster administrator creates `install-config.yaml` and adds the following in it. 
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: NLB
+      networkLoadBalancer:
+        eipAllocations:
+        - eipalloc-1111
+        - eipalloc-2222
+        - eipalloc-3333
+        - eipalloc-4444
+        - eipalloc-5555 
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
```

Comment:
Don't list pullSecret and sshKey here unless it is something new for EIP configuration.
```suggestion
...
```

Reply:
ok